### PR TITLE
Types Rewrite

### PR DESCRIPTION
**Pull request by noahrepublic**
Types rewrite fixing poor and incorrect typing. #31

### CODE REVIEW

**Comment by noahrepublic** [comment]
Feels good. Confirm all the types are working for you too, maybe my environment is glitched or something, studio sometimes takes a bit to update types when I am editing.

**Comment by FireBird702** [comment]
Fixed `CriticalState` not switching off, added `isInCriticalState` to `CriticalStateSignal`

Renamed `CriticalState` to `IsCriticalState`

`CriticalStateSignal` and `IssueSignal` are moved from `Store` to `DataKeep` because there was no purpose to keep them in `Store` so it might be a slightly breaking change

Switched from `ID` to `Id` (breaking changes anyway)

Added new `DataKeep` tab in API section

Updated FastSignal to `10.3.2`

`.GetStore()` now requires wrapper as a third parameter (can be set to `{}`)

Default Wrapper is accessible from `DataKeep.Wrapper`

**Comment by noahrepublic** [comment]
This is looking really good. I'm away from home right now so can't really review as well

File: docs/Versus.md [hunk 1]
```diff
@@ -9,4 +9,4 @@ sidebar_position: 6
 - Profile does not automatically clean up internal connections, making the developer have to perform inconvenient clean ups
 - ProfileService async calls make it difficult to wait for Profiles to be loaded. Causing weird patterns when waiting for Profiles, DataKeep is promise based
 - Shorter, cleaner, scripts for faster future development, and contributors (vs ProfileService fitting classes inside one script for micro-performance)
-- Type checking. There is one caveat due to Luau limitations, can not type check what Promises return (but shows on documentation)
+- Type checking
```
Comment (noahrepublic) [inline docs/Versus.md]: Maybe should still mention the limitations in promise typing, how we only get one layer

Comment (FireBird702) [reply docs/Versus.md]: Yeah, u are right. I had no idea for anything sensible

**Review by FireBird702 — APPROVED** [review]

**Pull request merged**